src/java.net.http/share/classes/java/net/http/HttpRequest.java

```diff
@@ -654,5 +656,37 @@ public static BodyPublisher ofByteArrays(Iterable<byte[]> iter) {
         public static BodyPublisher noBody() {
             return new RequestPublishers.EmptyPublisher();
         }
+
+        /**
+         * Creates a {@code BodyPublisher} that publishes a request
+         * body consisting of the aggregation of the bytes published
+         * by a sequence of publishers.
+         *
+         * <p> If the sequence is empty an {@linkplain #noBody() empty} publisher
+         * is returned. Otherwise each publisher is lazily subscribed to in turn,
+         * until all the body bytes are published, an error occurs, or the
+         * returned publisher's subscription is cancelled.
+         *
+         * <p> The request body published by the aggregate body publisher
+         * is logically equivalent to the request body that would have
+         * been published by aggregating all the bytes of each publisher
+         * in sequence. However, an implementation is free to split
+         * the sequence of bytes in any way it chooses.
+         *
+         * @implNote If the aggregate
+         * publisher's subscription is {@linkplain Flow.Subscription#cancel()
+         * cancelled}, or an error occurs while publishing the bytes, not all
+         * publishers in the sequence may be subscribed to.
+         * A publisher may be subscribed several times in sequence if the
```

Comment:
I think that this sentence can be moved out of the implNote and into normative text

Reply:
OK.